### PR TITLE
Disabling var in the typescript editor

### PR DESCRIPTION
**Pull request by abchatra**
We don't fully adhere to javascript var semantics. This change disables var as the way to define a variable and will throw an error requiring let or const.

Motivated by https://github.com/Microsoft/pxt/issues/744
The real issue in #744 is typescript binds to one global variable, but we implement let semantics for var which contradicts.

### CODE REVIEW

File: pxtlib/emitter/emitter.ts [hunk 1]
```diff
@@ -2525,9 +2525,16 @@ ${lbl}: .short 0xffff
         function emitBlock(node: Block) {
             node.statements.forEach(emit)
         }
+        function checkForLetOrConst(declList: VariableDeclarationList): boolean  {
+            if ((declList.flags & NodeFlags.Let) || (declList.flags & NodeFlags.Const)) {
+                return true;
+            }
+            throw userError(9203, lf("variable needs to be defined using 'let' or 'const'"));
```
Comment (riknoll) [inline pxtlib/emitter/emitter.ts]: I think we are already using the 9203 error code elsewhere.

**Comment by pelikhan** [comment]
We should automatically fix these issues by using "let" and notify the users that we did so. 

* A learner won't be able to really understand why "var" is not valid 
* we are giving a choice (let vs const) which is also confusing. Some students won't be able to decide between both.

**Comment by abchatra** [comment]
I am going to experiment with first one and see how it goes. I will update the PR after that
Agree with 2nd point. I am going to update with just a choice for let.

**Comment by pelikhan** [comment]
What abou issuing an error saying "use let instead of var"?

**Comment by abchatra** [comment]
Yes will fix this asap.

**Comment by abchatra** [comment]
Going to reopen an another one. This one is stale

**Pull request closed without merging**